tools/expotools/src/Changelogs.ts

```diff
@@ -101,11 +108,12 @@ export class Changelog {
    */
   async getVersionsAsync(): Promise<string[]> {
     const tokens = await this.getTokensAsync();
-    const versionTokens = tokens.filter(
-      (token) => token.type === Markdown.TokenType.HEADING && token.depth === VERSION_HEADING_DEPTH
-    ) as Markdown.HeadingToken[];
 
-    return versionTokens.map((token) => token.text);
+    return tokens
+      .filter<Markdown.HeadingToken>((token): token is Markdown.HeadingToken =>
+        isVersionToken(token)
+      )
```

Comment:
```suggestion
      .filter<Markdown.HeadingToken>(isVersionToken)
```
Do you need another (exactly the same) `type guard` here? 🤔

Reply:
Unfortunately yes 😞 `isVersionToken` may have the second argument so I can just put it directly as filter function. Also, in this case type guard is needed, without it TypeScript cannot infer properly the type of resulted array 🤷‍♂️

Reply:
Actually I could remove `<Markdown.HeadingToken>` and that's what I did just now. This type guard needs to be repeated tho.